libraries/DateTime/contracts/DateTime.aes

```diff
@@ -114,13 +62,181 @@ contract DateTime =
     let timestamp = timestamp + second  
     timestamp
 
-// Private functions and internal logic
+  public function parse_timestamp(timestamp : int) : date_time = 
+    let dt : date_time = init_date_time(timestamp)
+    
+    let dt : date_time = p_get_year(dt)
+    let dt : date_time = p_get_month(dt)
+    let dt : date_time = p_get_day(dt)
+    let dt : date_time = dt{hour = rest(timestamp / 60 /60, 24)}
+    let dt : date_time = dt{minute = rest(timestamp / 60, 60)}
+    let dt : date_time = dt{second = rest(timestamp, 60)}
+    let wd : int = rest(timestamp / state.day_in_seconds + 4, 7)
+    let dt : date_time = dt{weekday = wd}
+    dt
+
+  public function diff_years(from_timestamp : int, to_timestamp : int) : int = 
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let years : int = parse_timestamp(to_timestamp).year - parse_timestamp(from_timestamp).year
+    years
```

Comment:
can directly return (true for all occurrences in contract)

Reply:
done